rmgpy/data/thermo.py

```diff
@@ -1295,53 +1295,57 @@ def get_thermo_data(self, species, metal_to_scale_to=None, training_set=None):
+                    if original_molecule.get_radical_count() > quantum_mechanics.settings.maxRadicalNumber:
+                        # Too many radicals for direct calculation: use HBI.
+                        logging.info("{0} radicals on {1} exceeds limit of {2}. Using HBI method.".format(
+                            original_molecule.get_radical_count(),
+                            species.label,
+                            quantum_mechanics.settings.maxRadicalNumber,
+                        ))
+
+                        # Need to estimate thermo via each resonance isomer
+                        thermo = []
+                        for molecule in species.molecule:
+                            molecule.clear_labeled_atoms()
+                            # Try to see if the saturated molecule can be found in the libraries
+                            tdata = self.estimate_radical_thermo_via_hbi(molecule, self.get_thermo_data_from_libraries)
+                            priority = 1
+                            if tdata is None:
+                                # Then attempt quantum mechanics job on the saturated molecule
+                                tdata = self.estimate_radical_thermo_via_hbi(molecule, quantum_mechanics.get_thermo_data)
+                                priority = 2
+                            if tdata is None:
+                                # Fall back to group additivity
+                                tdata = self.estimate_thermo_via_group_additivity(molecule)
+                                priority = 3
+
+                            thermo.append((priority, tdata.get_enthalpy(298.), molecule, tdata))
+
+                        if len(thermo) > 1:
+                            # Sort thermo first by the priority, then by the most stable H298 value
+                            thermo = sorted(thermo, key=lambda x: (x[0], x[1]))
+                            for i, therm in enumerate(thermo):
+                                logging.debug("Resonance isomer {0} {1} gives H298={2:.0f} J/mol"
+                                              "".format(i + 1, therm[2].to_smiles(), therm[1]))
+                            # Save resonance isomers reordered by their thermo
+                            species.molecule = [item[2] for item in thermo]
+                            original_molecule = species.molecule[0]
+                        thermo0 = thermo[0][3]
+
+                        # update entropy by symmetry correction
+                        thermo0.S298.value_si -= constants.R * math.log(species.get_symmetry_number())
+
+                    else:  # Not too many radicals: do a direct calculation.
+                        thermo0 = quantum_mechanics.get_thermo_data(original_molecule)  # returns None if it fails
+            except ValueError: #rdkit fails to generate conformers 
+                logging.error("Quantum Mechanics calculation failed for species: {0}".format(species.label))
```

Comment:
Change the formatting here for performance. I had to remind myself of the correct syntax [here](https://github.com/ReactionMechanismGenerator/RMG-Py/issues/1776)

Reply:
Should be fixed now

Reply:
Hi, I just created a new environment and installed RMG just now. I am still stuck with gaussian. 

`Adding rate rules from training set in kinetics families...
Traceback (most recent call last):
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmg.py", line 118, in <module>
    main()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmg.py", line 112, in main
    rmg.execute(**kwargs)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/main.py", line 673, in execute
    self.initialize(**kwargs)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/main.py", line 496, in initialize
    self.load_database()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/main.py", line 420, in load_database
    family.add_rules_from_training(thermo_database=self.database.thermo)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/data/kinetics/family.py", line 1269, in add_rules_from_training
    reactant.thermo = thermo_database.get_thermo_data(reactant, training_set=True, metal_to_scale_to=metal)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/data/thermo.py", line 1344, in get_thermo_data
    thermo0 = quantum_mechanics.get_thermo_data(original_molecule)  # returns None if it fails
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/main.py", line 224, in get_thermo_data
    thermo0 = qm_molecule_calculator.generate_thermo_data()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 363, in generate_thermo_data
    self.qm_data = self.generate_qm_data()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/gaussian.py", line 254, in generate_qm_data
    if self.verify_output_file():
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/gaussian.py", line 170, in verify_output_file
    qm_data = self.parse()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 340, in parse
    qm_data = parse_cclib_data(cclib_data, radical_number + 1)  # Should `radical_number+1` be `self.molecule.multiplicity` in the next line of code? It's the electronic ground state degeneracy.
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/qmdata.py", line 101, in parse_cclib_data
    rotational_constants = (cclib_data.rotcons[-1], 'cm^-1')
IndexError: list index out of range`

Can you help me with this one?

Reply:
@amarkpayne was the cclib version change tested properly for qmtp with Gaussian?

Reply:
@mjohnson541  Actually I should have poster it on the earlier thread.. I just came to this one as a solution. I will post it on the cclib parse error thread as well.. **just for historic reasons**

Reply:
@mjohnson541 With the rdkit issue you identified and a qmfallback, you seem to have fixed the issue? 

I did get mopac and tried the same and I still stopped at 

```
Reading existing thermo file QMfiles/pm3/BRXCPUJQSHQGEU-UHFFFAOYSA-N.thermo
Traceback (most recent call last):
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmg.py", line 118, in <module>
    main()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmg.py", line 112, in main
    rmg.execute(**kwargs)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/main.py", line 923, in execute
    trimolecular_react=self.trimolecular_react)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/model.py", line 614, in enlarge
    self.apply_thermo_to_species(procnum)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/rmg/model.py", line 820, in apply_thermo_to_species
    quantum_mechanics.run_jobs(self.new_species_list, procnum=procnum)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/main.py", line 253, in run_jobs
    _write_qm_files_star(qm_arg)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/main.py", line 264, in _write_qm_files_star
    return _write_qm_files(*args)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/main.py", line 271, in _write_qm_files
    quantum_mechanics.get_thermo_data(mol)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/main.py", line 216, in get_thermo_data
    thermo0 = qm_molecule_calculator.generate_thermo_data()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 363, in generate_thermo_data
    self.qm_data = self.generate_qm_data()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/mopac.py", line 302, in generate_qm_data
    self.create_geometry()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 325, in create_geometry
    self.geometry.generate_rdkit_geometries()
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 135, in generate_rdkit_geometries
    self.save_coordinates_from_rdmol(rdmol, min_e_id, rdatom_idx)
  File "/glb/ap/bngsti/proj/ir6/inaveu/sw/rmg/RMG-Py/rmgpy/qm/molecule.py", line 173, in save_coordinates_from_rdmol
    point = rdmol.GetConformer(min_e_id).GetAtomPosition(atom.sorting_label)
ValueError: Bad Conformer Id
```

Reply:
Are you running in parallel? It looks like if you're running in parallel it calls different code and the error doesn't get caught.

Reply:
No, I am not running or experimenting right now with parallel options. I am just running it over the linux prompt over with  rmg.py input.py (unless there is something hidden inside of  input.py)...

Reply:
Just to reiterate that I am unable to run the 1,3-hexadiene example with mopac. I am stuck at same "ValueError: Bad Confomer ID). Is it working at your end?